hydromt_fiat/workflows/exposure_vector.py

```diff
@@ -538,52 +548,89 @@ def setup_ground_floor_height(
+            max_potential_damage, float
+        ):      
+                # modify the column manually 
+                #self.exposure_db[target_attribute] = max_potential_damage
+                NotImplemented
 
-            damage_values = preprocess_jrc_damage_values(damage_source, country)
-
-        elif max_potential_damage == "hazus_max_potential_damages":
-            damage_source = self.data_catalog.get_dataframe(max_potential_damage)
-            damage_values = preprocess_hazus_damage_values(damage_source)
-        else:
+        elif isinstance(max_potential_damage, str) or isinstance(
+            max_potential_damage, Path
+        ):
+            # A single file is used to assign the ground floor height to the assets
+            gfh = self.data_catalog.get_geodataframe(max_potential_damage)
+            gdf = self.get_full_gdf(self.exposure_db)
+            gdf = join_spatial_data(gdf, gfh, attr_name, method, max_dist, self.logger)
+            self.exposure_db = self._set_values_from_other_column(
+                gdf, target_attribute, attr_name
+            )
+        elif isinstance(max_potential_damage, list):
+            # Multiple files are used to assign the ground floor height to the assets
             NotImplemented
-            # TODO: Implement other ways of setting the max potential damage, e.g. from a
-            # vector file or from a table.
-
-        # Calculate the area of each object
-        gdf = self.get_full_gdf(self.exposure_db)[["Primary Object Type", "geometry"]]
-        gdf = get_area(gdf)
+            
 
-        # Set the damage values to the exposure data
-        if damage_types is None:
-            damage_types = ["total"]
+    def setup_ground_elevation(
+        self,
+        ground_elevation: Union[int, float, None, str, Path, List[str], List[Path]],
+        exposure_db: pd.DataFrame,
+        exposure_geoms: gpd.GeoDataFrame,
+    ) -> None:
 
-        for damage_type in damage_types:
-            # Calculate the maximum potential damage for each object and per damage type
-            try:
-                self.exposure_db[
-                    f"Max Potential Damage: {damage_type.capitalize()}"
-                ] = [
-                    damage_values[building_type][damage_type.lower()] * square_meters
-                    for building_type, square_meters in zip(
-                        gdf["Primary Object Type"], gdf["area"]
-                    )
-                ]
-            except KeyError as e:
-                self.logger.warning(
-                    f"Not found in the {max_potential_damage} damage "
-                    f"value data: {e}"
-                )
+        if ground_elevation:
+            # This function was developed by Willem https://github.com/interTwin-eu/DT-flood/blob/DemonstrationNotebooks/Notebooks/SetupFIAT.ipynb
+            #TODO: Find equivalent functions in hydromt
+            # Read in the DEM
+            dem = rasterio.open(ground_elevation)
```

Comment:
Can you put the code under `if ground_elevation:` for getting the ground elevation and adding it to the exposure objects in a separate function in gis.py?

Reply:
It is implemented now in gis.py